packages/opentelemetry-resources/test/Resource.test.ts

```diff
@@ -287,6 +287,116 @@ describe('Resource', () => {
     });
   });
 
+  describe('schema URL support', () => {
+    it('should create resource with schema URL', () => {
+      const schemaUrl = 'https://example.com/schema';
+      const resource = resourceFromAttributes({ attr: 'value' }, schemaUrl);
+
+      assert.strictEqual(resource.getSchemaUrl?.(), schemaUrl);
+    });
+
+    it('should create resource without schema URL', () => {
+      const resource = resourceFromAttributes({ attr: 'value' });
+
+      assert.strictEqual(resource.getSchemaUrl?.(), undefined);
+    });
+
+    it('should merge resources with schema URL priority given to other resource', () => {
+      const resource1 = resourceFromAttributes(
+        { attr1: 'value1' },
+        'https://schema1.com'
+      );
+      const resource2 = resourceFromAttributes(
+        { attr2: 'value2' },
+        'https://schema2.com'
+      );
+
+      const mergedResource = resource1.merge(resource2);
+
+      assert.strictEqual(
+        mergedResource.getSchemaUrl?.(),
+        'https://schema2.com'
+      );
+    });
+
+    it('should retain schema URL from base resource when other has no schema URL', () => {
+      const schemaUrl = 'https://example.com/schema';
+      const resource1 = resourceFromAttributes({ attr1: 'value1' }, schemaUrl);
+      const resource2 = resourceFromAttributes({ attr2: 'value2' });
+
+      const mergedResource = resource1.merge(resource2);
+
+      assert.strictEqual(mergedResource.getSchemaUrl?.(), schemaUrl);
+    });
+
+    it('should retain schema URL from the resource that has it when merging', () => {
+      const resource1 = resourceFromAttributes({ attr1: 'value1' }, '');
+      const resource2 = resourceFromAttributes(
+        { attr2: 'value2' },
+        'https://example.com/schema'
+      );
+
+      const mergedResource = resource1.merge(resource2);
+
+      assert.strictEqual(
+        mergedResource.getSchemaUrl?.(),
+        'https://example.com/schema'
+      );
+    });
+
+    it('should have empty schema URL when merging resources with no schema URL', () => {
+      const resource1 = resourceFromAttributes({ attr1: 'value1' }, '');
+      const resource2 = resourceFromAttributes({ attr2: 'value2' }, '');
+
+      const mergedResource = resource1.merge(resource2);
+
+      assert.strictEqual(mergedResource.getSchemaUrl?.(), undefined);
+    });
+
+    it('should handle merging with empty string schema URLs', () => {
+      const resource1 = resourceFromAttributes({ attr1: 'value1' }, '');
+      const resource2 = resourceFromAttributes(
+        { attr2: 'value2' },
+        'https://valid.schema'
```

Comment:
please try to use TLDs intended for testing wherever possible (`.test` or `.example`)